src/IROperator.h

```diff
@@ -1909,6 +1908,28 @@ Expr saturating_cast(Expr e) {
  * maximum values of the result type. */
 EXPORT Expr saturating_cast(Type t, Expr e);
 
+/** Prevents the simplifier from changing floating-point operations
+ * within the argument expression while returning the argument as its result.
+ * Can be used to ensure stricter floating-point semantics but does not by itself
+ * affect lower-level code generation. Mostly provided for debugging. */
+inline Expr no_fp_simplify(Expr e) {
+    Type t = e.type();
+    return Internal::Call::make(t, Internal::Call::no_fp_simplify,
+                                {std::move(e)}, Internal::Call::Intrinsic);
```

Comment:
PureIntrinsic? Not actually sure about that... This is 'meta-side-effecty' but I don't think it's side effecting in the way that violates the meaning of pure that is used to determine if simplifications/optimizations are valid.